Games/Minesweeper/Field.cpp

```diff
@@ -271,15 +272,40 @@ void Field::reset()
     set_updates_enabled(true);
 }
 
+void Field::flood_mark(Square& square)
```

Comment:
Can we share some of this code with `on_square_clicked()` in a nice way?
There's a bit of duplication here.